src/main/resources/static/theme-toggle.js

```diff
@@ -1,11 +1,14 @@
 (function () {
   const darkModeMediaQuery = window.matchMedia('(prefers-color-scheme: dark)')
 
+  const initialTheme = darkModeMediaQuery.matches || localStorage.getItem('darkMode') === 'true'
```

Comment:
```suggestion
  const initialTheme = localStorage.getItem('darkMode') === 'true'
```
This fixed it for me :)